backend/src/main/java/com/example/skicurort/curort/CurortService.java

```diff
@@ -0,0 +1,44 @@
+package com.example.skicurort.curort;
+
+import com.example.skicurort.exception.NoIdException;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CurortService {
+
+  private final CurortRepo curortRepo;
+
+  public void save(Curort curort) {
+    curortRepo.save(curort);
+  }
+
+  public List<Curort> getAll() {
+    return curortRepo.findAll();
+  }
+
+  public Curort findById(Long id) throws NoIdException {
+    return curortRepo.findById(id).orElseThrow(() -> new NoIdException(id));
+  }
+
+  public void delete(Long id) throws NoIdException {
+    if (!curortRepo.existsById(id)) {
+      throw new NoIdException(id);
+    }
+    curortRepo.deleteById(id);
+  }
+
+  public void updateDto(CurortDTO curortDTO, Long id) throws NoIdException {
+    if (curortRepo.existsById(id)) {
+      curortRepo.findById(id).get().setCurortName(curortDTO.curortName());
+      curortRepo.findById(id).get().setCurortAdress(curortDTO.curortAdress());
+      curortRepo.findById(id).get().setCurortPhonenumber(curortDTO.curortPhonenumber());
+      curortRepo.findById(id).get().setCurrortEmail(curortDTO.currortEmail());
+
+    } else {
+      throw new NoIdException(id);
+    }
```

Comment:
1. nie ma potrzeby wyciągać findById tyle razy
2. nigdzie nie zapisujesz tego co wyciągnąłeś więc tam metoda nie zadziała
3. link do podobnej metody z mojego repo, myślę, że może Ci pomóc to ogarnąć i uprościć.
 https://github.com/KasiaCw/menu-wizard/blob/main/backend/src/main/java/pl/jcw/menuwizard/recipe/RecipeService.java#L38

Reply:
Docelowo miałem tak to ogarniete ale musiałem tej zmiany nie skomitować i gdzieś jak dzialaismy z gitem to mi to znikło. Czy tak by też mogło być jeżeli w dto nie mam id

 public Curort updateDto(CurortDTO curortDTO, Long id) throws NoIdException {

    return curortRepo
        .findById(id)
        .map(
            curort -> {
              curort.setCurortName(curortDTO.curortName());
              curort.setCurortAdress(curortDTO.curortAdress());
              curort.setCurortPhonenumber(curortDTO.curortPhonenumber());
              curort.setCurrortEmail(curortDTO.currortEmail());
              return curortRepo.save(curort);
            })
        .orElseThrow(() -> new NoIdException(id));
  }

Reply:
tak jest ok tylko na koniec zmapuj jeszcze do DTO tak  jak rozmawialiśmy tutaj https://github.com/Ski-Curort/ski-curort/pull/18#discussion_r1009285204

Reply:
Zrobione